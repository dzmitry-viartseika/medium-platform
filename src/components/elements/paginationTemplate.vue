<template>
  <ul class="app-pagination">
    <li v-for="page in pages" :key="page" class="app-pagination__item"
    :class="{'app-pagination__item_active': currentPage === page}"
    >
      <router-link :to="{path: url, query: {page: page}}" tag="div">
        {{ page }}
      </router-link>
    </li>
  </ul>
</template>

<script>
export default {
  name: 'paginationTemplate',
  props: {
    total: {
      type: Number,
      required: true,
    },
    limit: {
      type: Number,
      required: true,
    },
    currentPage: {
      type: Number,
      required: true,
    },
    url: {
      type: String,
      required: true,
    },
  },
  watch: {
    currentPage() {
      this.$emit('update', 10, this.currentPage);
    },
  },
  computed: {
    pages() {
      const pagesCount = Math.ceil(this.total / this.limit);
      return [...Array(pagesCount).keys()].map((el) => el + 1);
    },
  },
};
</script>

<style scoped lang="scss">
@import "./src/assets/scss/variables";

.app-pagination {
  display: flex;
  list-style-type: none;

  &__item {
    padding: 5px;
    cursor: pointer;
    border: 1px solid $colorBtn;
    transition: opacity .15s ease-in;
    margin: 3px;

    &_active {
      background: $colorBtn;
      color: $color-white;
    }

    &:hover {
      opacity: .8;
    }
  }
}
</style>
