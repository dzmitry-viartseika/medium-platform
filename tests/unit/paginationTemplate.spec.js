import { createLocalVue, mount } from '@vue/test-utils';
import paginationTemplate from '@/components/elements/paginationTemplate.vue';
import VueRouter from 'vue-router';

const localVue = createLocalVue();
localVue.use(VueRouter);
const router = new VueRouter();

describe('headerTemplate.vue', () => {
  test('snapshot', () => {
    const wrapper = mount(paginationTemplate, {
      router,
      propsData: {
        total: 500,
        limit: 10,
        currentPage: 10,
        url: '/',
      },
    });
    expect(wrapper.find('.app-pagination')).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
