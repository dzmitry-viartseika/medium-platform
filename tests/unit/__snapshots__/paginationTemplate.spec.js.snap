// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headerTemplate.vue snapshot 1`] = `
<ul class="app-pagination">
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      1
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      2
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      3
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      4
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      5
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      6
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      7
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      8
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      9
    </router-link>
  </li>
  <li class="app-pagination__item app-pagination__item_active">
    <router-link to="[object Object]" tag="div">
      10
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      11
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      12
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      13
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      14
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      15
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      16
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      17
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      18
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      19
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      20
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      21
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      22
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      23
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      24
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      25
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      26
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      27
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      28
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      29
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      30
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      31
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      32
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      33
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      34
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      35
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      36
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      37
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      38
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      39
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      40
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      41
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      42
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      43
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      44
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      45
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      46
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      47
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      48
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      49
    </router-link>
  </li>
  <li class="app-pagination__item">
    <router-link to="[object Object]" tag="div">
      50
    </router-link>
  </li>
</ul>
`;
